src/transformers/models/eomt/modular_eomt.py

```diff
@@ -0,0 +1,1157 @@
+
+        # Expand attention mask to 4d mask.
+        if attention_mask is not None:
+            attention_mask = attention_mask[:, None, ...].expand(-1, self.num_heads, -1, -1)
+            attention_mask = attention_mask.to(dtype=query_states.dtype)
+
+        attention_interface: Callable = eager_attention_forward
+        if self.config._attn_implementation != "eager":
+            attention_interface = ALL_ATTENTION_FUNCTIONS[self.config._attn_implementation]
+
+        attn_output, attn_weights = attention_interface(
+            self,
+            query_states,
+            key_states,
+            value_states,
+            attention_mask,
+            dropout=0.0 if not self.training else self.attention_dropout,
+            scaling=self.scaling,
+            is_causal=self.is_causal,
+            **kwargs,
+        )
+
+        attn_output = attn_output.reshape(batch_size, q_len, self.embed_dim)
+
+        attn_output = self.proj_out(attn_output)
+        attn_output = self.proj_drop(attn_output)
+
+        output = (attn_output, attn_weights) if output_attentions else (attn_output, None)
+
+        return output
+
+
+class EoMTMLP(Dinov2MLP):
+    pass
+
+
+class EoMTLayerScale(Dinov2LayerScale):
+    pass
+
+
+class EoMTDropPath(Dinov2DropPath):
+    pass
+
+
+class EoMTLayer(nn.Module):
+    def __init__(self, config: EoMTConfig) -> None:
+        super().__init__()
+
+        self.norm1 = nn.LayerNorm(config.hidden_size, eps=config.layer_norm_eps)
+        self.attention = EoMTAttention(config)
+        self.layer_scale1 = EoMTLayerScale(config)
+        self.drop_path = EoMTDropPath(config.drop_path_rate) if config.drop_path_rate > 0.0 else nn.Identity()
+
+        self.norm2 = nn.LayerNorm(config.hidden_size, eps=config.layer_norm_eps)
+        self.mlp = EoMTMLP(config)
+        self.layer_scale2 = EoMTLayerScale(config)
+
+    def forward(
+        self,
+        hidden_states: torch.Tensor,
+        attention_mask: Optional[torch.Tensor] = None,
+        output_attentions: bool = False,
+    ) -> Union[Tuple[torch.Tensor, torch.Tensor], Tuple[torch.Tensor]]:
+        normalized_hidden_states = self.norm1(hidden_states)
+        self_attention_outputs = self.attention(
+            normalized_hidden_states,
+            attention_mask=attention_mask,
+            output_attentions=output_attentions,
+        )
+        attention_output = self_attention_outputs[0]
+        attention_output = self.layer_scale1(attention_output)
+        outputs = self_attention_outputs[1:]
+
+        hidden_states = self.drop_path(attention_output) + hidden_states
+
+        layer_output = self.norm2(hidden_states)
+        layer_output = self.mlp(layer_output)
+        layer_output = self.layer_scale2(layer_output)
+
+        layer_output = self.drop_path(layer_output) + hidden_states
+
+        outputs = (layer_output,) + outputs
+
+        return outputs
```

Comment:
We can still use modular like so
```suggestion
class EoMTLayer(Dinov2Layer):
    def __init__(self, config: EoMTConfig) -> None:
        Dinov2Layer().__init__()

        self.norm1 = nn.LayerNorm(config.hidden_size, eps=config.layer_norm_eps)
        self.attention = EoMTAttention(config)
        self.layer_scale1 = EoMTLayerScale(config)
        self.drop_path = EoMTDropPath(config.drop_path_rate) if config.drop_path_rate > 0.0 else nn.Identity()

        self.norm2 = nn.LayerNorm(config.hidden_size, eps=config.layer_norm_eps)
        self.mlp = EoMTMLP(config)
        self.layer_scale2 = EoMTLayerScale(config)
```
It's ok if we have head_mask imo, especially as it's provided as attention_mask is provided as an arg in `EoMTForUniversalSegmentation`, so you don't need to change anything